betterproto/tests/inputs/xfail.py

```diff
@@ -0,0 +1,10 @@
+# Test cases that are expected to fail, e.g. unimplemented features or bug-fixes.
+# Remove from list when fixed.
+tests = {
```

Comment:
It's not obvious to me what's the point of this centralised approach (and the overhead) over just applying the xfail markers where they're needed?

Reply:
adding markers to an individual test is easy, using `@pytest.mark.xfail`, but adding it to a specific test-case within `@pytest.mark.parametrize("test_case_name", test_cases)` requires changing `test_cases` as shown [here](https://docs.pytest.org/en/latest/skipping.html#skip-xfail-with-parametrize).

```
@pytest.mark.parametrize(
    ("n", "expected"),
    [
        (1, 2),
        pytest.param(1, 0, marks=pytest.mark.xfail),
        pytest.param(1, 3, marks=pytest.mark.xfail(reason="some bug")),
        (2, 3),
        (3, 4),
        (4, 5),
        pytest.param(
            10, 11, marks=pytest.mark.skipif(sys.version_info >= (3, 0), reason="py2k")
        ),
    ],
)
```
basically, it means that the list of test-cases needs to be hardcoded, instead of auto-generated from the directories. when hardcoding the test-case list, the xfail marks also need to be repeated for every test function individually.

Reply:
so the TestCases class generates those lists, and takes the ones that should be marked as xfail from a list. I've externalized the list because I expect it to be frequently updated, and it was hard to explain where to do it when it somewhere inside a constructor arg.

Reply:
gotcha. fair enough.